app/hooks/access_control_attributes.ts

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useCallback, useEffect, useRef, useState} from 'react';
+
+import {fetchChannelAccessControlAttributes} from '@actions/remote/channel_access_control_attributes';
+import {useServerUrl} from '@context/server';
+
+// Module-level cache for access control attributes
+// The cache stores data with a timestamp to implement a TTL (time-to-live)
+const attributesCache: Record<string, {
+    data: Record<string, string[]>;
+    timestamp: number;
+}> = {};
+
+// Cache TTL in milliseconds (5 minutes)
+const CACHE_TTL = 5 * 60 * 1000;
+
+/**
+ * A hook for fetching access control attributes for a channel.
+ *
+ * @param entityType - The type of entity (e.g., 'channel')
+ * @param entityId - The ID of the entity
+ * @param hasPolicyEnforced - Whether the entity has policy enforcement enabled
+ * @returns An object containing the attribute tags, loading state, and fetch function
+ */
+export const useAccessControlAttributes = (
+    entityType: 'channel',
+    entityId: string | undefined,
+    hasPolicyEnforced: boolean | undefined,
+) => {
+    const serverUrl = useServerUrl();
+    const [attributeTags, setAttributeTags] = useState<string[]>([]);
+    const [loading, setLoading] = useState(false);
+    const [error, setError] = useState<Error | null>(null);
+
+    // Use useRef to track if the component is mounted
+    const isMounted = useRef(true);
+
+    // Cleanup on unmount
+    useEffect(() => {
+        return () => {
+            isMounted.current = false;
+        };
+    }, []);
+
+    // Helper function to process attribute data and extract tags
+    const processAttributeData = useCallback((data: Record<string, string[]> | undefined) => {
+        if (!data || typeof data !== 'object') {
+            return [];
+        }
+
+        const tags: string[] = [];
+
+        // Process each entry in the attributes object
+        for (const values of Object.values(data)) {
+            if (Array.isArray(values)) {
+                for (const value of values) {
+                    if (value !== undefined && value !== null) {
+                        tags.push(`${value}`);
+                    }
+                }
+            }
+        }
+
+        return tags;
+    }, []);
+
+    const fetchAttributes = useCallback(async (forceRefresh = false) => {
+        if (!entityId || !hasPolicyEnforced) {
+            return;
+        }
+
+        setLoading(true);
+        setError(null);
+
+        try {
+            // Check cache first
+            const cacheKey = `${serverUrl}:${entityType}:${entityId}`;
+            const cachedEntry = attributesCache[cacheKey];
+            const now = Date.now();
+
+            // Use cache if it exists and is not too old and forceRefresh is false
+            if (!forceRefresh && cachedEntry && (now - cachedEntry.timestamp < CACHE_TTL)) {
```

Comment:
Same comment about cache hit as in webapp PR -- https://github.com/mattermost/mattermost/pull/30971#discussion_r2083827557